src/main/java/com/gregtechceu/gtceu/api/item/tool/GTToolType.java

```diff
@@ -194,9 +194,151 @@ public class GTToolType {
+        .toolTag(TagUtil.createItemTag("shovels", true))
+        .harvestTag(TagUtil.createBlockTag("mineable/pickaxe", true))
+        .harvestTag(TagUtil.createBlockTag("mineable/shovel", true))
+        .toolStats(b -> b.blockBreaking().aoe(1, 1, 2)
+                .attackDamage(1.0F).attackSpeed(-3.2F).durabilityMultiplier(4.0F)
+                .brokenStack(ToolHelper.SUPPLY_POWER_UNIT_MV)
+                /*.behaviors(TorchPlaceBehavior.INSTANCE)*/)
+        .sound(GTSoundEntries.DRILL_TOOL, true)
+        .electric(GTValues.MV)
+        .build();
+    public static final GTToolType DRILL_HV = GTToolType.builder("hv_drill")
+        .idFormat("hv_%s_drill")
+        .toolTag(TagUtil.createItemTag("tools/drills", false))
+        .toolTag(TagUtil.createItemTag("pickaxes", true))
+        .toolTag(TagUtil.createItemTag("shovels", true))
+        .harvestTag(TagUtil.createBlockTag("mineable/pickaxe", true))
+        .harvestTag(TagUtil.createBlockTag("mineable/shovel", true))
+        .toolStats(b -> b.blockBreaking().aoe(2, 2, 4)
+                .attackDamage(1.0F).attackSpeed(-3.2F).durabilityMultiplier(5.0F)
+                .brokenStack(ToolHelper.SUPPLY_POWER_UNIT_HV)
+                /*.behaviors(TorchPlaceBehavior.INSTANCE)*/)
+        .sound(GTSoundEntries.DRILL_TOOL, true)
+        .electric(GTValues.HV)
+        .build();
+    public static final GTToolType DRILL_EV = GTToolType.builder("ev_drill")
+        .idFormat("ev_%s_drill")
+        .toolTag(TagUtil.createItemTag("tools/drills", false))
+        .toolTag(TagUtil.createItemTag("pickaxes", true))
+        .toolTag(TagUtil.createItemTag("shovels", true))
+        .harvestTag(TagUtil.createBlockTag("mineable/pickaxe", true))
+        .harvestTag(TagUtil.createBlockTag("mineable/shovel", true))
+        .toolStats(b -> b.blockBreaking().aoe(3, 3, 6)
+                .attackDamage(1.0F).attackSpeed(-3.2F).durabilityMultiplier(6.0F)
+                .brokenStack(ToolHelper.SUPPLY_POWER_UNIT_EV)
+                /*.behaviors(TorchPlaceBehavior.INSTANCE)*/)
+        .sound(GTSoundEntries.DRILL_TOOL, true)
+        .electric(GTValues.EV)
+        .build();
+    public static final GTToolType DRILL_IV = GTToolType.builder("iv_drill")
+        .idFormat("iv_%s_drill")
+        .toolTag(TagUtil.createItemTag("tools/drills", false))
+        .toolTag(TagUtil.createItemTag("pickaxes", true))
+        .toolTag(TagUtil.createItemTag("shovels", true))
+        .harvestTag(TagUtil.createBlockTag("mineable/pickaxe", true))
+        .harvestTag(TagUtil.createBlockTag("mineable/shovel", true))
+        .toolStats(b -> b.blockBreaking().aoe(4, 4, 8)
+                .attackDamage(1.0F).attackSpeed(-3.2F).durabilityMultiplier(7.0F)
+                .brokenStack(ToolHelper.SUPPLY_POWER_UNIT_IV)
+                /*.behaviors(TorchPlaceBehavior.INSTANCE)*/)
+        .sound(GTSoundEntries.DRILL_TOOL, true)
+        .electric(GTValues.IV)
+        .build();
```

Comment:
This is fine for now, but I think these could be refactored into a single array maybe?

Reply:
could maybe make a helper at some point, but an array? no thank you